### PR TITLE
Use Unicode \u2192 in Java source for rightarrow

### LINKED ISSUE #890: Problem with the symbols "Floating point to Integer" and "Integer to Floating point"
image

Icons and symbols of arithmetic components "Floating point to Integer" and "Integer to Floating point" are not very satisfactory. 

Test under windows 10   language FR-CH

### PR DESCRIPTION
**Pull request by simoc**
Use Unicode `\u2192` instead of `→` in Java source code for rightarrow symbol.

The Java source code for other Arithmetic components already uses this standard `\uNNNN` notation.

### CODE REVIEW

**Comment by MarcinOrlowski** [comment]
Butq why exactly? And why rightone only?

**Comment by BFH-ktt1** [comment]
@simoc : Could you please make the linter happy :-)

**Comment by MarcinOrlowski** [comment]
Aside from the linter I'd like to know why we are doing this change in first place :)

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski :
Because in the drawing of the icon a construct like this one:
```TextLayout l = new TextLayout("A", f, g2.getFontRenderContext());```
Is used, where the string needs to be unicode encoded (UTF-8 gives bizarre results).
Maybe there is a way to use UTF-8, but I do not know how, so I think this is a proper solution.

File: src/main/java/com/cburch/logisim/std/arith/FPToInt.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public FPToInt() {
         new Object[] {BitWidth.create(8), BitWidth.create(32), ROUND_OPTION});
     setKeyConfigurator(new BitWidthConfigurator(StdAttr.FP_WIDTH));
     setOffsetBounds(Bounds.create(-40, -20, 40, 40));
-    setIcon(new ArithmeticIcon("FP→I", 2));
+    setIcon(new ArithmeticIcon("FP\u2192I", 2));
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/arith/FPToInt.java:84]: Please add comment explaining why using `\u` is required here. Or, maybe, also add a issue ticket (low pri) and link to it from code comment.

File: src/main/java/com/cburch/logisim/std/arith/FPToInt.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public void paintInstance(InstancePainter painter) {
 
     g.setColor(Color.GRAY);
     painter.drawPort(IN);
-    painter.drawPort(OUT, "F→I", Direction.WEST);
+    painter.drawPort(OUT, "F\u2192I", Direction.WEST);
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/arith/FPToInt.java:103]: Please add comment explaining why using `\u` is required here.

File: src/main/java/com/cburch/logisim/std/arith/IntToFP.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ public void paintInstance(InstancePainter painter) {
 
     g.setColor(Color.GRAY);
     painter.drawPort(IN);
-    painter.drawPort(OUT, "I→F", Direction.WEST);
+    painter.drawPort(OUT, "I\u2192F", Direction.WEST);
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/arith/IntToFP.java:102]: Please add comment explaining why using `\u` is required here.

File: src/main/java/com/cburch/logisim/std/arith/IntToFP.java [hunk 4]
```diff
@@ -80,7 +80,7 @@ public IntToFP() {
         new Object[] {BitWidth.create(8), BitWidth.create(32), SIGNED_OPTION});
     setKeyConfigurator(new BitWidthConfigurator(StdAttr.FP_WIDTH));
     setOffsetBounds(Bounds.create(-40, -20, 40, 40));
-    setIcon(new ArithmeticIcon("I→FP", 2));
+    setIcon(new ArithmeticIcon("I\u2192FP", 2));
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/arith/IntToFP.java:83]: Please add comment explaining why using `\u` is required here.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/std/arith/FPToInt.java (hunk 1 above)
Comment (simoc) [reply src/main/java/com/cburch/logisim/std/arith/FPToInt.java:84]: I replaced inline Unicode right arrow characters in the Java source code with the `\uNNNN` syntax.

The `\uNNNN` syntax is the correct way to use Unicode characters in Java source code and there are many instances of this in the logisim source code, including other Arithmetic components, and `\u2192` is already used for a right arrow character in the Java class `ShiftRegister` and `PlaRomPanel`.

**Review by BFH-ktt1 — APPROVED** [review]

**Comment by MarcinOrlowski** [comment]
I requested some changes previously. This is still not addressed, hence lack of approval.

**Comment by MarcinOrlowski** [comment]
I am still go a feeling you masked the problem instead of fixing he issue. Do we know what was the issue here? Anyone knows **why exactly** the issue happened? I am pretty surprised there's any issue with UTF-8 strings. So we need to understand what exact value we have in both cases and compare.

**Review by maehne — APPROVED** [review]
@simoc: Thanks for your fix to issue #890! 

@MarcinOrlowski: I agree that understanding the root of the issue why UTF-8 is not properly handled would be good. Feel free to open a separate issue to track this and later remove `\u????` occurences in the code base. Nevertheless, it should not prevent merging this PR and I don't see much point in adding comments to these occurences. Then, it should properly done also in other parts of the source code.

@BFH-ktt1: I therefore propose to merge this PR as it is 

I have observed similar issues in the past with Unicode characters in the localisation files, which caused issues depending on the used JDK version. In some way, it is to be expected from a language and its associated tools, which are now around for 30 years. UTF-8 was not yet around in the beginning.

**Review by MarcinOrlowski — APPROVED** [review]
So while I do not think this PR really fixes the culprit of #890, let it merge but please keep the ticket opened.

**Pull request merged**